tests/integration/test_s3.py

```diff
@@ -871,5 +872,53 @@ def test_can_set_utf_8_headers(self):
                         Bucket=bucket_name, Key="foo.txt")
 
 
+class TestSupportedPutObjectBodyTypes(BaseS3ClientTest):
+    def setUp(self):
+        super(TestSupportedPutObjectBodyTypes, self).setUp()
+        self.region = 'us-east-1'
+        self.client = self.create_client()
+        self.bucket_name = self.create_bucket()
+        self.tempdir = tempfile.mkdtemp()
```

Comment:
We could also move ``create_client`` method to ``BaseS3ClientTest`` class and update its ``setUp`` method.